crates/lightbeam/src/module.rs

```diff
@@ -563,15 +563,17 @@ pub fn translate_only(data: &[u8]) -> Result<TranslatedModule, Error> {
         let memories = section.get_memory_section_reader()?;
         let mem = translate_sections::memory(memories)?;
 
-        assert_le!(
-            mem.len(),
-            1,
-            "Multiple memory sections not yet unimplemented"
-        );
+        if mem.len() > 1 {
+            return Err(Error::Input(
+                "Multiple memory sections not yet unimplemented".to_string(),
+            ));
+        }
 
         if !mem.is_empty() {
             let mem = mem[0];
-            assert_eq!(Some(mem.limits.initial), mem.limits.maximum);
+            if Some(mem.limits.initial) != mem.limits.maximum {
+                return Err(Error::Input("Memory limits not matching".to_string()));
```

Comment:
It'd be good for the error message here to indicate that while this condition is valid in wasm, lightbeam doesn't yet support it.

Reply:
i replaced by "Custom memory limits not supported in lightbeam"